### PR TITLE
fix: fix asset-list e2e test

### PR DESCRIPTION
**Pull request by salimtb**
## **Description**
fix e2e test for the unified list

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/28822?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. run `PORTFOLIO_VIEW=1 yarn build:test`
2. Run `PORTFOLIO_VIEW=1 yarn test:e2e:single --browser=chrome test/e2e/tests/multichain/asset-list.spec.ts --debug --leave-running`

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by salimtb** [comment]
The error occurs because we attempt to find the TST token created on chain ID 1337, which belongs to the testnet section. However, the unified list excludes tokens from the testnet network, causing the test to fail. To resolve this, I updated the test to include Polygon in the network list, so the TST token is now created on Polygon.

File: test/e2e/tests/multichain/asset-list.spec.ts [hunk 1]
```diff
@@ -146,20 +148,20 @@ describe('Multichain Asset List', function (this: Suite) {
...
           toastText,
-          `You're now using ${LOCALHOST}`,
+          `You're now using ${POLYGON_NAME_MAINNET}`,
           'Toast text is correct',
         );
       },
     );
   });
-  it('shows correct asset and balance when swapping on a different chain', async function () {
+  it.only('shows correct asset and balance when swapping on a different chain', async function () {
```
Comment (sahar-fehri) [inline test/e2e/tests/multichain/asset-list.spec.ts:164]: Can we remove this `only`

File: test/e2e/tests/multichain/asset-list.spec.ts [hunk 2]
```diff
@@ -14,13 +14,15 @@ import AssetListPage from '../../page-objects/pages/asset-list';
 const NETWORK_NAME_MAINNET = 'Ethereum Mainnet';
 const LINEA_NAME_MAINNET = 'Linea Mainnet';
 const LOCALHOST = 'Localhost 8545';
+const POLYGON_NAME_MAINNET = 'Polygon';
 const BALANCE_AMOUNT = '24.9956';
 
-function buildFixtures(title: string) {
+function buildFixtures(title: string, chainId: number = 1337) {
   return {
```
Comment (sahar-fehri) [inline test/e2e/tests/multichain/asset-list.spec.ts:21]: I'm still not sure how this used to work before without issues 🤔

File: test/e2e/tests/multichain/asset-list.spec.ts (hunk 1 above)
Comment (salimtb) [reply test/e2e/tests/multichain/asset-list.spec.ts]: done [28ae946](https://github.com/MetaMask/metamask-extension/pull/28822/commits/28ae94600bfcfee55c5f970b1c2f9ecc28ad5cb7)

File: test/e2e/tests/multichain/asset-list.spec.ts (hunk 2 above)
Comment (salimtb) [reply test/e2e/tests/multichain/asset-list.spec.ts:21]: It appears that the feature flag was not enabled in the CI environment, which is the primary reason we didn't encounter this issue earlier.

**Review by NidhiKJha — CHANGES_REQUESTED** [review]
LGTM! Tests are passing. Just left one small comment

File: test/e2e/tests/multichain/asset-list.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (salimtb) [reply test/e2e/tests/multichain/asset-list.spec.ts]: done [28ae946](https://github.com/MetaMask/metamask-extension/pull/28822/commits/28ae94600bfcfee55c5f970b1c2f9ecc28ad5cb7)

**Review by NidhiKJha — DISMISSED** [review]
LGTM

**Review by sahar-fehri — DISMISSED** [review]

**Review by NidhiKJha — APPROVED** [review]

**Review by sahar-fehri — APPROVED** [review]

**Pull request merged**